### PR TITLE
Feature(appliance): db connection pinger and validator

### PR DESCRIPTION
**Pull request by DaedalusG**
Introduce a pinger for the appliance to poll the various Sourcegraph dbs and ensure connection is possible given the current connection envvar configuration. 

## Test plan

Unit Tests

### CODE REVIEW

File: internal/appliance/upgrades/util.go [hunk 1]
```diff
@@ -47,3 +52,45 @@ func DetermineUpgradePolicy(currentVersion, targetVersion string) (downtime bool
...
+		f = connections.RawNewFrontendDB
+	case "codeintel":
+		f = connections.RawNewCodeIntelDB
+	case "codeinsights":
+		f = connections.RawNewCodeInsightsDB
+	}
+
+	fmt.Printf("Checking connection to %s database...\n", name)
+
+	if db, err := f(obsvCtx, dsn, "appliance"); err != nil {
```
Comment (Chickensoupwithrice) [inline internal/appliance/upgrades/util.go:84]: Minor nit: All the `f`'s being defined here are confusing to me. I'm also not sure that the `func` is necessary, or why it's passed all those parameters that it doesn't use?

**Comment by Chickensoupwithrice** [comment]
Hell yeah dude! ~ 
Asides from the minor nit, LGTM 🥳

File: internal/appliance/upgrades/util.go (hunk 1 above)
Comment (DaedalusG) [reply internal/appliance/upgrades/util.go:84]: Oh this is just a functional programming thing, its just defining f as a func that takes those param, then we assign the corresponding DB handle func and call it later

Comment (Chickensoupwithrice) [reply internal/appliance/upgrades/util.go:84]: Oh, I understand now. I misread the `var` definition. I think a better name than `f` would still be a good idea.

**Review by Chickensoupwithrice — APPROVED** [review]

File: internal/appliance/upgrades/util.go (hunk 1 above)
Comment (DaedalusG) [reply internal/appliance/upgrades/util.go]: Nice changed it to `connect`

**Pull request merged**